model_utils.py

```diff
@@ -5,7 +5,9 @@
 
 logger = logging.getLogger(__name__)
 
-def get_local_huggingface_tokenizer_model(model_name, model_path=None, dtype=None):
+def get_local_huggingface_tokenizer_model(model_name, model_path=None, dtype=None, auth_token=None):
+    use_auth_token=True if auth_token else None
```

Comment:
This line is odd since True and None are different types. This might be better.

```suggestion
    use_auth_token=auth_token is not None
```

Reply:
Hmm, you're right. They're different types, I'll make this change. I tried you suggestion, but it doesn't work when `auth_token=False`

Reply:
Fixed. I tested `use_auth_token=False` and it works on public models.